src/arm/phyboard.c

```diff
@@ -0,0 +1,672 @@
+        b->pins[31].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 };
+    }
+    b->pins[31].gpio.pinmap = (3*32 + 9);
+    b->pins[31].gpio.parent_id = 0;
+    b->pins[31].gpio.mux_total = 0;
+    b->pins[31].uart.mux_total = 0;
+
+    strncpy(b->pins[32].name, "X_MMC2_DAT3", MRAA_PIN_NAME_SIZE);
+    b->pins[32].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    /* PIN-Number "33" with name "X_UART2_TX_GPIO3_10" */
+    if (uart2_enabled == 1) {
+        strncpy(b->pins[33].name, "X_UART2_TX", MRAA_PIN_NAME_SIZE);
+        b->pins[33].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 };
+    } else {
+        strncpy(b->pins[33].name, "X_GPIO3_10", MRAA_PIN_NAME_SIZE);
+        b->pins[33].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 };
+    }
+    b->pins[33].gpio.pinmap = (3*32 + 10);
+    b->pins[33].gpio.parent_id = 0;
+    b->pins[33].gpio.mux_total = 0;
+    b->pins[33].uart.mux_total = 0;
+    
+    strncpy(b->pins[34].name, "GND", MRAA_PIN_NAME_SIZE);
+    b->pins[34].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    /* PIN-Number "35" with name "X_UART3_RX_GPIO2_18" */
+    if (uart3_enabled == 1) {
+        strncpy(b->pins[35].name, "X_UART3_RX", MRAA_PIN_NAME_SIZE);
+        b->pins[35].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 };
+    } else {
+        strncpy(b->pins[35].name, "X_GPIO2_18", MRAA_PIN_NAME_SIZE);
+        b->pins[35].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 };
+    }
+    b->pins[35].gpio.pinmap = (2*32 + 18);
+    b->pins[35].gpio.parent_id = 0;
+    b->pins[35].gpio.mux_total = 0;
+    b->pins[35].uart.mux_total = 0;
+
+    /*  PIN-Number "36" with name "X_UART3_TX_GPIO2_19" */
+    if (uart3_enabled == 1) {
+        strncpy(b->pins[36].name, "X_UART3_TX", MRAA_PIN_NAME_SIZE);
+        b->pins[36].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 };
+    } else {
+        strncpy(b->pins[36].name, "X_GPIO2_19", MRAA_PIN_NAME_SIZE);
+        b->pins[36].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 };
+    }
+    b->pins[36].gpio.pinmap = (2*32 + 19);
+    b->pins[36].gpio.parent_id = 0;
+    b->pins[36].gpio.mux_total = 0;
+    b->pins[36].uart.mux_total = 0;
+
+    /* PIN-Number "37" with name "X_INTR1_GPIO0_20" */
+    strncpy(b->pins[37].name, "X_GPIO0_20", MRAA_PIN_NAME_SIZE);
+    b->pins[37].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 0 };
+    b->pins[37].gpio.pinmap = (0*32 + 20);
+    b->pins[37].gpio.parent_id = 0;
+    b->pins[37].gpio.mux_total = 0;
+
+    strncpy(b->pins[38].name, "X_GPIO0_7", MRAA_PIN_NAME_SIZE);
+    b->pins[38].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 0 };
+    b->pins[38].gpio.pinmap = (0*32 + 7);
+    b->pins[38].gpio.parent_id = 0;
+    b->pins[38].gpio.mux_total = 0;
+
+    strncpy(b->pins[39].name, "X_AM335_EXT_WAKEUP", MRAA_PIN_NAME_SIZE);
```

Comment:
`MRAA_PIN_NAME_SIZE` is 12 and `b->pins[39].name` is allocated to 12 symbols, the name won't fit in full and will lack the null-terminator